src/main/java/se/sundsvall/seabloader/service/mapper/InvoiceMapper.java

```diff
@@ -57,4 +74,30 @@ public static InExchangeInvoiceStatusType toInExchangeInvoice(final String xml)
 		final var unmarshaller = JAXBContext.newInstance(InExchangeInvoiceStatusType.class).createUnmarshaller();
 		return (InExchangeInvoiceStatusType) JAXBIntrospector.getValue(unmarshaller.unmarshal(xmlSource));
 	}
+
+	private static InvoicePdf toInvoicePdf(final InExchangeInvoiceStatusType inExchangeInvoiceStatusType) {
+		final var attachments = inExchangeInvoiceStatusType.getAttachments();
+
+		if (Objects.isNull(attachments)) {
+			LOGGER.warn("No attachments found in invoice with invoiceId: {}", inExchangeInvoiceStatusType.getInvoice().getInvoiceId());
+			return null;
+		}
+
+		final var invoicePdfMerger = new InvoicePdfMerger();
+		final var name = inExchangeInvoiceStatusType
+			.getAttachments()
+			.getAttachment().stream()
+			.findFirst()
+			.map(InExchangeInvoiceStatusTypeAttachment.Attachment::getName)
+			.orElse(null);
+
+		if (Objects.isNull(name)) {
+			LOGGER.warn("No attachments found in invoice with invoiceId: {}", inExchangeInvoiceStatusType.getInvoice().getInvoiceId());
+			return null;
+		}
+
+		return new InvoicePdf()
+			.content(Base64.getEncoder().encodeToString(invoicePdfMerger.mergePdfs(inExchangeInvoiceStatusType).toString().getBytes(Charset.defaultCharset())))
```

Comment:
Is `toString()` really the correct way to transform an outputStream into a String?
Have you tested this?

Reply:
Yes, and it seems to work.